internal/allocator/allocator.go

```diff
@@ -350,10 +350,8 @@ func (a *Allocator) isPoolCompatibleWithService(p *config.Pool, svc *v1.Service)
 // service has no IP allocated, "" is returned.
 func (a *Allocator) Pool(svc string) string {
 	if alloc := a.allocated[svc]; alloc != nil {
-		pool := poolFor(a.pools.ByName, alloc.ips)
-		if pool != nil {
-			return pool.Name
-		}
+		// Should never be an allocation without a pool.
```

Comment:
Let's remove this comment. I think it's safe to assume this is always the case, so the comment will just add friction to the reader who doesn't know the previous version and will doubt there are extra cases.

If we don't completely trust alloc always containing a pool we might consider logging or even panicking.

Reply:
will remove it

Reply:
done